pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go

```diff
@@ -248,3 +249,66 @@ func TestNodeResourcesLeastAllocated(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeResourcesLeastAllocatedWithWeight(t *testing.T) {
+	cpuAndMemory := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("1000m"),
+						v1.ResourceMemory: resource.MustParse("2000"),
+					},
+				},
+			},
+			{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("2000m"),
+						v1.ResourceMemory: resource.MustParse("3000"),
+					},
+				},
+			},
+		},
+	}
+	tests := []struct {
+		pod          *v1.Pod
+		pods         []*v1.Pod
+		nodes        []*v1.Node
+		argsRaw      string
+		expectedList framework.NodeScoreList
+		name         string
+	}{
+		{
+			// CPU Score: ((4000 - 3000) *100) / 4000 = 25
+			// Memory Score: ((10000 - 5000) *100) / 10000 = 50
+			// Node1 Score: (25 * 1 + 50 * 2) / (1 + 2) = 41
+			// CPU Score: ((6000 - 3000) *100) / 6000 = 50
+			// Memory Score: ((10000 - 5000) *100) / 10000 = 50
+			// Node2 Score: (50 * 1 + 50 * 2) / (1 + 2) = 50
+			pod:          &v1.Pod{Spec: cpuAndMemory},
+			nodes:        []*v1.Node{makeNode("machine1", 4000, 10000), makeNode("machine2", 6000, 10000)},
+			argsRaw:      `{"resources" : [{"name" : "memory", "weight" : 2}, {"name" : "cpu", "weight" : 1}]}`,
```

Comment:
Is it possible to just expand `TestNodeResourcesLeastAllocated` with this test case?

Reply:
done